tissuumaps/static/js/utils/regionUtils.js

```diff
@@ -1798,24 +1798,44 @@ regionUtils.JSONToRegions= function(filepath){
         if (path != null) {
             filepath = path + "/" + filepath;
         }
-        fetch(filepath)
-        .then((response) => {
-            return response.json();
-        })
-        .then((regionsobj) => {
-            regionUtils.JSONValToRegions(regionsobj);
-        });
+        if (filepath.includes(".pbf")) {
+            // Load GeoJSON stored in Geobuf format (https://github.com/mapbox/geobuf)
+            fetch(filepath)
+            .then((response) => {
+                return response.arrayBuffer();
+            })
+            .then((regionsobj) => {
+                const data = new Pbf(regionsobj);
+                regionUtils.JSONValToRegions(geobuf.decode(data));
+            });
+        } else if (filepath.includes(".geojson") || filepath.includes(".json")) {
+            fetch(filepath)
+            .then((response) => {
+                return response.json();
+            })
+            .then((regionsobj) => {
+                regionUtils.JSONValToRegions(regionsobj);
+            });
+        }
     }
     else if(window.File && window.FileReader && window.FileList && window.Blob) {
         var op=tmapp["object_prefix"];
         var text=document.getElementById(op+"_region_files_import");
         var file=text.files[0];
         var reader = new FileReader();
-        reader.onload=function(event) {
-            // The file's text will be printed here
-            regionUtils.JSONValToRegions(JSON.parse(event.target.result));
-        };
-        reader.readAsText(file);
+        if (file.name.includes(".pbf")) {
+            // Load GeoJSON stored in Geobuf format (https://github.com/mapbox/geobuf)
+            reader.onload=function(event) {
+                const data = new Pbf(event.target.result);
+                regionUtils.JSONValToRegions(geobuf.decode(data));
+            };
+            reader.readAsArrayBuffer(file);
+        } else if (file.name.includes(".geojson") || file.name.includes(".json")) {
+            reader.onload=function(event) {
+                regionUtils.JSONValToRegions(JSON.parse(event.target.result));
+            };
+            reader.readAsText(file);
+        }
```

Comment:
Would be good to have an `else` clause here, warning of unsupported file types

Reply:
Thanks, added it now!